packages/cozy-client/src/models/applications.js

```diff
@@ -0,0 +1,69 @@
+import get from 'lodash/get'
+const STORE_SLUG = 'store'
+
+/**
+ * Return Store URL where an app/konnector can be installed / updated
+ *
+ * @param  {Array}  [appData=[]]   Apps data, as returned by endpoint /apps/ or
+ * /konnectors/
+ * @param  {object} [app={}] AppObject
+ * @returns {string}                URL as string
+ */
+export const getStoreInstallationURL = (appData = [], app = {}) => {
+  if (!app.slug) {
+    throw new Error('Expected app / konnector with the defined slug')
+  }
+
+  const storeApp = isInstalled(appData, { slug: STORE_SLUG })
+  if (!storeApp) return null
+
+  const storeUrl = storeApp.links && storeApp.links.related
+
+  if (!storeUrl) return null
+
+  return `${storeUrl}#/discover/${app.slug}/install`
+}
+
+/**
+ *
+ * @param {Array} apps Array of apps returned by /apps /konnectors
+ * @param {object} wantedApp io.cozy.app with at least a slug
+ * @returns {object} The io.cozy.app is installed or undefined if not
+ */
+export const isInstalled = (apps = [], wantedApp = {}) => {
```

Comment:
The signature looks weird and the implementation looks like it is very easy to make a mistake a have a O(n2) behavior (since to check installation status we have to iterate on all the list. Keying by slug would solve this problem IMHO).

Reply:
`Keying by slug?` what do you mean by that? How can I access to apps....slug===slug? 

Do you suggest that instead of having a simple Array apps[0], apps[1], we had something `keyed` like apps[slug]?

Reply:
I think that if apps were instead an appsBySlug object, it would be harder to shoot yourself in the foot.

Reply:
https://github.com/cozy/cozy-libs/issues/875